tests/unit-tests/EnvironmentTest.php

```diff
@@ -62,6 +66,48 @@ public function identifies_hostname()
         $this->assertEquals($identified->fqdn, $this->environment->hostname()->fqdn);
     }
 
+    /**
+     * @test
+     */
+    public function maintenance_test()
+    {
+        try {
+            $middleware = new HostnameActions(null, app()->make(Redirector::class));
+        } catch (\TypeError $ex) {
```

Comment:
Split this test method up and use phpdoc expectedException to anticipate an exception